timestamp.go

```diff
@@ -32,18 +32,14 @@ func (t Timestamp) MarshalJSON() ([]byte, error) {
 // a UNIX timestamp string is converted to int which is used for the Timestamp
 // object value
 func (t *Timestamp) UnmarshalJSON(b []byte) error {
-	ts, err := strconv.Atoi(string(b))
+	if len(b) == 0 {
+		return nil
+	}
+	ts, err := strconv.ParseFloat(string(b), 64)
 	if err != nil {
 		return err
 	}
-
-	int64ts := int64(ts)
-	if len(b) > 10 {
-		//support for milisecond timestamps
-		int64ts = int64(ts / 1000)
-	}
-	*t = Timestamp(time.Unix(int64ts, 0).UTC())
-
+	*t = Timestamp(time.Unix(int64(ts/1000), int64(ts)%1000*int64(time.Millisecond)).UTC())
```

Comment:
I think we need to maintain backward compatibility here.
So if we get an old timestamp from another MS that is using the old version of this library we need to convert it the old way.
So maybe something like 
if len(b) > 10 {
*t = Timestamp(time.Unix(int64(ts/1000), int64(ts)%1000*int64(time.Millisecond)).UTC())
} else {
*t = Timestamp(time.Unix(int64ts, 0).UTC())
}
?